File: sponge-jython/examples/script/py/processor_inheritance.py
"""
Sponge Knowledge base
Processor inheritance
"""

def onInit():
    # Variables for assertions only
    sponge.setVariable("result", None)

class AbstractEchoAction(Action):
    def calculateResult(self):
        return 1

class EchoAction(AbstractEchoAction):
    def onCall(self):
        return self.calculateResult() * 2

def onStartup():
    result = sponge.call("EchoAction")
    sponge.setVariable("result", result)
    sponge.logger.debug("Action returned: {}", result)

File: sponge-jruby/examples/script/rb/processor_inheritance.rb
# Sponge Knowledge Base
# Processor inheritance

def onInit
    # Variables for assertions only
    $sponge.setVariable("result", nil)
end

class AbstractEchoAction < Action
    def calculateResult()
        return 1
    end
end

class EchoAction < AbstractEchoAction
    def onCall()
        return calculateResult() * 2
    end
end

def onStartup
    result = $sponge.call("EchoAction")
    $sponge.setVariable("result", result)
    $sponge.logger.debug("Action returned: {}", result)
end
